File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    let answer = [];
    
    function hanoi(n, from, by, to) {
        let tmp = [];
        
        if(n === 1) {
            tmp.push(from);
            tmp.push(to);      
            answer.push(tmp);
            return;
        }
        
        hanoi(n - 1, from, to, by);
        
        tmp.push(from);
        tmp.push(to);
        answer.push(tmp);
        
        hanoi(n - 1, by, from, to);
    }
    
    hanoi(n, 1, 2, 3);
    
    return answer;
}

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<vector<int>> answer;

void hanoi(int cnt, int n, int from, int by, int to) {
    vector<int> tmp;
    
    if(n == 1) {
        tmp.push_back(from);
        tmp.push_back(to);
        answer.push_back(tmp);
        return;
    }

    hanoi(cnt + 1, n - 1, from, to, by); 
    
    tmp.push_back(from);
    tmp.push_back(to);
    answer.push_back(tmp);
    
    hanoi(cnt + 1, n - 1, by, from, to); 
}

vector<vector<int>> solution(int n) {      
    hanoi(0, n, 1, 2, 3);
    
    return answer;
}